src/Components/ProvisioningWizard/index.js

```diff
@@ -48,35 +54,32 @@ const ProvisioningWizard = ({ isOpen, onClose, image, ...props }) => {
     setStepIdReached((prevID) => (prevID < id ? id : prevID));
   };
 
-  return isOpen ? (
+  return (
     <WizardProvider>
-      <APIProvider>
-        <Wizard
-          {...props}
-          title="Launch"
-          description={`Launch image ${image.name}`}
-          steps={steps}
-          isOpen
-          onClose={onWizardClose}
-          onNext={onNext}
-          className={'provisioning'}
-          footer={<CustomFooter />}
-        />
-        <ConfirmModal isOpen={isConfirming} onConfirm={onCustomClose} onCancel={() => setConfirming(false)} />
-      </APIProvider>
+      <Wizard
+        {...props}
+        title="Launch"
+        description={`Launch image ${image.name}`}
+        steps={steps}
+        isOpen
+        onClose={onWizardClose}
+        onNext={onNext}
+        className={'provisioning'}
+        footer={<CustomFooter />}
+      />
+      <ConfirmModal isOpen={isConfirming} onConfirm={onCustomClose} onCancel={() => setConfirming(false)} />
     </WizardProvider>
-  ) : null;
+  );
 };
 
+const ProvisioningWizardWrapper = (props) => <APIProvider>{props.isOpen && <ProvisioningWizard {...props} />}</APIProvider>;
```

Comment:
nit? Wouldn't it be better to spilt this into a container that cosumes the wizard component wrapped by the providers (context and API) ?